### PR TITLE
feat(kdropdownmenu): new kongponent [beta]

### PR DESCRIPTION
**Pull request by kaiarrowood**
# Summary

## Vue 3

**Did you create a corresponding PR to add this change to the `beta` branch? (required)**

- [ ] **Yes**, here is a link to the PR: `[link to beta branch PR]`
- [ ] **No**, I did not create a PR for the `beta` branch and I have added the `port to beta branch` tag to this PR.
- [ ] **No**, because `[type your reasons]`

If you have questions, tag `@adamdehaven` or `@kaiarrowood`.

---

## PR Checklist

- [x] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

**Comment by adamdehaven** [comment]
@kaiarrowood is this one ready for review?

**Comment by kaiarrowood** [comment]
> @kaiarrowood is this one ready for review?

@adamdehaven It is now. Thanks for all the help!

File: docs/components/dropdown-menu.md [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+
+**KDropdownMenu** is a button (or any slotted content) that is clicked to trigger a menu popover beneath&nbsp;it.
+
+<KDropdownMenu label="Documentation" :items="deepClone(defaultItemsUnselected)" />
+
+## Props
+
+### label
+
+The label for the menu.
```
Comment (adamdehaven) [inline docs/components/dropdown-menu.md:13]: Should this say `menu trigger`?
```suggestion
The label for the menu trigger.
```

File: docs/components/dropdown-menu.md [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+
+### items
+
+An array of item objects containing a `label` property and other optional properties which will render a menu of [`KDropdownItems`](#KDropdownItem) .
```
Comment (adamdehaven) [inline docs/components/dropdown-menu.md:17]:
```suggestion
An array of objects containing a required `label` property and other optional properties which will render a menu of [`KDropdownItems`](#KDropdownItem) .
```

File: src/components/KDropdownMenu/KDropdownMenu.vue [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+import { defineComponent, PropType, ref, watch, onMounted, Ref } from 'vue'
+import KButton from '@/components/KButton/KButton.vue'
+import Kooltip from '@/components/KTooltip/KTooltip.vue'
+import KPop from '@/components/KPop/KPop.vue'
+import KToggle from '@/components/KToggle'
+import KDropdownItem from './KDropdownItem.vue'
+
+export interface DropdownItem {
+  label: string
+  // to?: RawLocation
```
Comment (adamdehaven) [inline src/components/KDropdownMenu/KDropdownMenu.vue:81]: Does`RawLocation` need to be removed?

Comment (kaiarrowood) [reply src/components/KDropdownMenu/KDropdownMenu.vue]: Fixed

**Review by adamdehaven — APPROVED** [review]
LGTM 🚀

**Pull request merged**